aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -300,6 +300,12 @@ public class AwsProperties implements Serializable {
 
   public static final String DYNAMODB_TABLE_NAME_DEFAULT = "iceberg";
 
+  public static final String DYNAMODB_V2_SCHEMA_DEFAULT_TABLE_NAME = "iceberg_dynamodb_v2";
+
+  public static final String DYNAMODB_CATALOG_SCHEMA_VERSION = "dynamodb.catalog.schema-version";
+
+  public static final int DYNAMODB_DEFAULT_SCHEMA_VERSION = 1;
```

Comment:
I am thinking if we should just make it default to v2, since there is not much users with prod dependencies of the Dynamo catalog as of now. Any thoughts? @munendrasn @amogh-jahagirdar

Reply:
AFAIK, Until now, some of the bugs encountered in DynamoDbCatalog might not surface to user depending on how users might be using the Catalog 
* listTables is uncommon operation, and only at higher cardinality the bug would [surface](https://github.com/apache/iceberg/commit/e1a09e78efd41bc8dbb095879254a89213b80464)
* default warehouse bug too might not surface if user always provides explicit location to table or warehouse path is not set in the namespace

Probably, there might be users with dependencies on DynamoDbCatalog. So, we would need to consider the impact for those users.
Still, it might be okay to make v2 as default if we call this change out in the release as major change - just a cautionary note

Reply:
Cool, let's keep it in v1 then, and see how this goes, and flip the default after 1-2 releases.